fdbbackup/backup.actor.cpp

```diff
@@ -3703,36 +3704,34 @@ int main(int argc, char* argv[]) {
 			}
 		}
 
-		for (auto k = knobs.begin(); k != knobs.end(); ++k) {
+		IKnobCollection::setGlobalKnobCollection(IKnobCollection::Type::CLIENT, Randomize::NO, IsSimulated::NO);
+		auto& g_knobs = IKnobCollection::getMutableGlobalKnobCollection();
+		for (const auto& [knobName, knobValueString] : knobs) {
 			try {
-				if (!globalFlowKnobs->setKnob(k->first, k->second) &&
-				    !globalClientKnobs->setKnob(k->first, k->second)) {
-					fprintf(stderr, "WARNING: Unrecognized knob option '%s'\n", k->first.c_str());
-					TraceEvent(SevWarnAlways, "UnrecognizedKnobOption").detail("Knob", printable(k->first));
-				}
+				auto knobValue = g_knobs.parseKnobValue(knobName, knobValueString);
+				g_knobs.setKnob(knobName, knobValue);
 			} catch (Error& e) {
 				if (e.code() == error_code_invalid_option_value) {
 					fprintf(stderr,
```

Comment:
Another good place for a `TEST` macro (might be caught in another location that checks for invalid knobs, so you might not need this one).

Reply:
I added a `TEST` macro for a similar case elsewhere.